### PR TITLE
Added interstitial_example to samples

### PR DESCRIPTION
**Pull request by malandr2**
## Description

Added `interstitial_example` project to the `samples` folder.

`interstitial_example` provides a countdown timer that fires an alert when the timer reaches 0. The interstitial ad is shown after the user hits "OK" on the alert dialog. User selects "Play Again" to repeat the process.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/googleads/googleads-mobile-flutter/issues). Indicate, which of these issues are resolved or fixed by this PR.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/googleads/googleads-mobile-flutter/issues
[Contributor Guide]: https://github.com/googleads/googleads-mobile-flutter/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: samples/admob/interstitial_example/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      print('InterstitialAd failed to load: $error');
+    },
+  ));
+ ```
+
+### Handle ad events
+The sample shows how to handle interstitial ad events.
+
+```
+ad.fullScreenContentCallback = FullScreenContentCallback(
```
Comment (jjliu15) [inline samples/admob/interstitial_example/README.md:41]:
```suggestion
_interstitialAd?.fullScreenContentCallback = FullScreenContentCallback(
```

File: samples/admob/interstitial_example/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                  ad.dispose();
+                },
+                // Called when a click is recorded for an ad.
+                onAdClicked: (ad) {});
+
+            // Keep a reference to the ad so you can show it later.
+            _interstitialAd = ad;
+          },
+          // Called when an ad request failed.
+          onAdFailedToLoad: (LoadAdError error) {},
```
Comment (jjliu15) [inline samples/admob/interstitial_example/lib/main.dart:114]: Maybe print out the error?

**Review by jjliu15 — APPROVED** [review]
Could you also add a github workflow that checks that this project builds? Can also be done in a separate PR.

**Review by luo86 — APPROVED** [review]

File: samples/admob/interstitial_example/lib/main.dart (hunk 2 above)
Comment (malandr2) [reply samples/admob/interstitial_example/lib/main.dart]: Done.

**Pull request merged**